### PR TITLE
added auth action types

### PR DESCRIPTION
**Pull request by ivanjermakov**

### CODE REVIEW

File: client/src/actions/AuthActionTypes.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+export interface LogoutSuccess {
+  type: typeof LOGOUT_SUCCESS;
+}
+
+export interface RegisterFail {
+  type: typeof REGISTER_FAIL;
+  payload: { message: Msg, status: number };
+}
+
+export type AuthActionTypes =
```
Comment (ivanjermakov) [inline client/src/actions/AuthActionTypes.ts]: Should split these definitions into two separate domain parts: `Login` and `Register` (`Logout` too, probably).

File: client/src/types/AuthForm.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface AuthForm {
```
Comment (ivanjermakov) [inline client/src/types/AuthForm.ts]: It probably should be `AuthState` or something similar.

File: client/src/reducers/authReducer.ts [hunk 3]
```diff
@@ -16,12 +14,11 @@ const initialState = {
   user: null
 };
 
-export default function (state = initialState, action: any) {
+export default function (state = initialState, action: AuthActionTypes) {
```
Comment (ivanjermakov) [inline client/src/reducers/authReducer.ts:16]: Do we really need these reducers at all? Maybe we should delegate state locally somehow (in `handleSubmit` functions for example). I feel like we will need to change state depending on action result too. Or we can do it separately?

File: client/src/actions/AuthActionTypes.ts (hunk 1 above)
Comment (mil-lesya) [reply client/src/actions/AuthActionTypes.ts]: It will be problematic because I pass all these types to the `authReducer`

Comment (ivanjermakov) [reply client/src/actions/AuthActionTypes.ts]: So we can split it to `authReducer` and `registerReducer`, what's the problem?

Comment (mil-lesya) [reply client/src/actions/AuthActionTypes.ts]: Okay, let's do so

File: client/src/types/AuthForm.ts (hunk 2 above)
Comment (mil-lesya) [reply client/src/types/AuthForm.ts]: Ok, i will fix

File: client/src/reducers/authReducer.ts (hunk 3 above)
Comment (mil-lesya) [reply client/src/reducers/authReducer.ts:16]: In this I don't know how best

Comment (ivanjermakov) [reply client/src/reducers/authReducer.ts:16]: So let's keep as it is for now

**Pull request merged**